### PR TITLE
samba: Add documentation for addon_configs

### PR DESCRIPTION
**Pull request by frenck**
SSIA

As mentioned here:

<https://github.com/home-assistant/addons/pull/3312#issuecomment-1813327616>

### CODE REVIEW

File: samba/DOCS.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ This addon exposes the following directories over smb (samba):
 Directory | Description
 -- | --
 `addons` | This is for your local add-ons.
+`addon_configs` | This is for your the configuration files or your add-ons.
```
Comment (agners) [inline samba/DOCS.md:28]:
```suggestion
`addon_configs` | This is for your add-on's configuration files.
```

Comment (frenck) [reply samba/DOCS.md:28]: That implies singular, which it isn't?

Comment (agners) [reply samba/DOCS.md:28]: Right, I think this should do it.

```suggestion
`addon_configs` | This is for your add-ons' configuration files.
```

Comment (agners) [reply samba/DOCS.md:28]: Or I guess the of form which might be nicer? Either way works for me.

```suggestion
`addon_configs` | This is for the configuration files of your add-ons.
```

**Review by agners — APPROVED** [review]

**Pull request merged**